src/js/components/Select/utils.js

```diff
@@ -17,7 +17,25 @@ export const getOptionLabel = (index, options, labelKey) =>
 export const getOptionValue = (index, options, valueKey) =>
   applyKey(options[index], valueKey);
 
-export const useDisabled = (disabled, disabledKey, options, valueKey) =>
+export const getOptionIndex = (options, i) => {
+  if (options) {
+    if (typeof i === 'object')
+      return options.findIndex((x) => JSON.stringify(x) === JSON.stringify(i));
```

Comment:
Is there any possibility that the key/value pairs of the object might be ordered differently between `x` and `i`? Wondering if a `deepEquals` type function would be more reliable than JSON.stringify

Reply:
Good point! Would you be okay if I created a separate issue for this instead of handling it in this PR? I think the JSON.stringify will cover the 80% case, but I agree it would be nice to have a more robust method for object comparison. I'm thinking the `deepEquals` type of function will take a bit of time writing and testing to make sure it is working as expected and feel like it would be better to create a separate issue rather than push back the release

Reply:
I'm wondering if this is being too strict. For instance, if the object in options has something unrelated to the value, like the label changes. I think we should be using valueKey for comparison whenever possible when we have object options.

Reply:
That's fair, I will go back and make that change

Reply:
that feels better, thanks